src/pools/D3MAaveDaiPlan.sol

```diff
@@ -0,0 +1,122 @@
+    TargetTokenLike public immutable variableDebt;
+
+    constructor(address dai_, address pool_) public D3MPlanBase(dai_, pool_) {
+
+        // Fetch the reserve data from Aave
+        (,,,,,,,, address stableDebt_, address variableDebt_, address interestStrategy_,) = LendingPoolLike(pool_).getReserveData(address(dai_));
+        require(stableDebt_ != address(0), "D3MAaveDai/invalid-stableDebt");
+        require(variableDebt_ != address(0), "D3MAaveDai/invalid-variableDebt");
+        require(interestStrategy_ != address(0), "D3MAaveDai/invalid-interestStrategy");
+
+        stableDebt = TargetTokenLike(stableDebt_);
+        variableDebt = TargetTokenLike(variableDebt_);
+        interestStrategy = InterestRateStrategyLike(interestStrategy_);
+    }
+
+    // --- Math ---
+    function _add(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x + y) >= x, "D3MAaveDai/overflow");
+    }
+    function _sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x - y) <= x, "D3MAaveDai/underflow");
+    }
+    function _mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x, "D3MAaveDai/overflow");
+    }
+    uint256 constant RAY  = 10 ** 27;
+    function _rmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, y) / RAY;
+    }
+    function _rdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, RAY) / y;
+    }
+
+    function maxBar() public override view returns (uint256) {
+        return interestStrategy.getMaxVariableBorrowRate();
+    }
+
+    // --- Automated Rate targeting ---
+    function calculateTargetSupply(uint256 targetInterestRate) public view returns (uint256) {
+        uint256 base = interestStrategy.baseVariableBorrowRate();
+        require(targetInterestRate > base, "D3MAaveDai/target-interest-base");
+        require(targetInterestRate <= maxBar(), "D3MAaveDai/above-max-interest");
+
+        // Do inverse calculation of interestStrategy
+        uint256 variableRateSlope1 = interestStrategy.variableRateSlope1();
+        uint256 targetUtil;
+        if (targetInterestRate > _add(base, variableRateSlope1)) {
+            // Excess interest rate
+            uint256 r = targetInterestRate - base - variableRateSlope1;
+            targetUtil = _add(_rdiv(_rmul(interestStrategy.EXCESS_UTILIZATION_RATE(), r), interestStrategy.variableRateSlope2()), interestStrategy.OPTIMAL_UTILIZATION_RATE());
+        } else {
+            // Optimal interest rate
+            targetUtil = _rdiv(_rmul(_sub(targetInterestRate, base), interestStrategy.OPTIMAL_UTILIZATION_RATE()), variableRateSlope1);
+        }
+        return _rdiv(_add(stableDebt.totalSupply(), variableDebt.totalSupply()), targetUtil);
+    }
+
+    function calcSupplies(uint256 availableAssets) external override view returns(uint256 totalAssets, uint256 targetAssets) {
```

Comment:
Neat - here and in many other places we do `returns(..` instead of `returns (...`.
This is while in most places we do have the space.
Suggest doing search and replace.